### PR TITLE
2023.9.x feature add per page resource list

### LINKED ISSUE #1497: Add per page dropdown to resource lists

### PR DESCRIPTION
**Pull request by EmmyMay**
# The following changes are implemented
TODO: Summary

# Changes in the user interface:
TODO: Add screenshots, recordings or remove this section

# Checklist when submitting a final (!draft) PR
 - [x] Commits are tidied up, squashed if needed and follow guidelines in CONTRIBUTING.md
 - [x] Code builds
 - [x] All existing tests pass
 - [x] All new critical code is covered by tests
 - [x] PR is linked to the relevant issue(s)
 - [x] Rebased with the target branch
closes #1497

### CODE REVIEW

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 1]
```diff
@@ -129,11 +129,21 @@
...
-        {{ getPagination }}
+      <div class="d-flex gap-1 align-items-center">
+        <div
+          v-if="!hideTotal"
+          class="text-nowrap"
+        >
+          {{ getPagination }}
+        </div>
+        
+        <div>
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Where is the text like in recordList?

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 2]
```diff
@@ -282,6 +292,7 @@ export default {
     return {
       selected: [],
       selectableItemIDs: [],
+      rowsPerPage: 10,
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Dont think we need this

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 3]
```diff
@@ -408,6 +433,11 @@ export default {
       this.$router.replace({ query: { ...this.$route.query, page, pageCursor } })
     },
 
+    loadRows () {
+      this.$router.replace({ query: { ...this.$route.query, page: 1, limit: this.pagination.limit } })
+      this.$refs.resourceList.refresh()
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue:455]: This shouldn't be needed

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 4]
```diff
@@ -408,6 +433,11 @@ export default {
       this.$router.replace({ query: { ...this.$route.query, page, pageCursor } })
     },
 
+    loadRows () {
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: You need to use the value you select, not the pagination limit

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 5]
```diff
@@ -129,11 +129,21 @@
...
+        <div
+          v-if="!hideTotal"
+          class="text-nowrap"
+        >
+          {{ getPagination }}
+        </div>
+        
+        <div>
+          <b-form-select
+            v-model="pagination.limit"
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Dont v-modal pagination.limit, the value should be updated by the query.replace. You only pass the value here.

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 6]
```diff
@@ -129,11 +129,21 @@
...
+            :options="perPageOptions"
+            @change="loadRows"
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Lets call this by what it is please, like you did in recordList, onPerPageChange

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 7]
```diff
@@ -129,11 +129,21 @@
...
+      <div class="d-flex gap-1 align-items-center">
+        <div
+          v-if="!hideTotal"
+          class="text-nowrap"
+        >
+          {{ getPagination }}
+        </div>
+        
+        <div>
+          <b-form-select
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: This should be togglable in a prop.
You need to amend showFooter aswell just like you did in recordList. Try it with different options, how does it look and work.

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 8]
```diff
@@ -129,11 +129,21 @@
       v-if="showFooter"
       #footer
     >
-      <div
-        v-if="!hideTotal"
-        class="text-nowrap"
-      >
-        {{ getPagination }}
+      <div class="d-flex gap-1 align-items-center">
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: It should wrap like recordList, styles should be similar. User shouldn't see a difference

**Review by Fajfa — CHANGES_REQUESTED** [review]

File: client/web/workflow/src/components/Configurator/basic.vue [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div>
     <b-form-group
-      :label="$t('general:label')"
+      :label="$t('general:label.label')"
```
Comment (Fajfa) [inline client/web/workflow/src/components/Configurator/basic.vue:4]: Double check this is not used anywhere else

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 10]
```diff
@@ -126,65 +126,72 @@
     </b-card-body>
 
     <template
-      v-if="showFooter"
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Why has this been removed?

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 11]
```diff
@@ -126,65 +126,72 @@
...
+          </div>
         
-        <div>
-          <b-form-select
-            v-model="pagination.limit"
-            :options="perPageOptions"
-            @change="loadRows"
-          />
+          <div class="d-flex flex-wrap gap-1 align-items-center">
+            <div>
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: We dont need this div here

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 12]
```diff
@@ -433,8 +446,8 @@ export default {
       this.$router.replace({ query: { ...this.$route.query, page, pageCursor } })
     },
 
-    loadRows () {
-      this.$router.replace({ query: { ...this.$route.query, page: 1, limit: this.pagination.limit } })
+    lhandlePerPageChange (v) {
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: lhandle?

Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: call it limit so you dont have to define it as limit: v

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 13]
```diff
@@ -126,65 +126,72 @@
...
-          />
+          <div class="d-flex flex-wrap gap-1 align-items-center">
+            <div>
+                {{ $t('general:resourceList.pagination.recordsPerPage') }}
+            </div>
+
+            <b-form-select
+              :options="perPageOptions"
+              :value="pagination.limit"
+              style="width: 70px;"
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Do we need this, if so why is it not in recordList?

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 14]
```diff
@@ -126,65 +126,72 @@
...
-            @change="loadRows"
-          />
+          <div class="d-flex flex-wrap gap-1 align-items-center">
+            <div>
+                {{ $t('general:resourceList.pagination.recordsPerPage') }}
+            </div>
+
+            <b-form-select
+              :options="perPageOptions"
+              :value="pagination.limit"
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue:149]: :value
:options

value is always priority, same reason v-model is always first

**Review by Fajfa — CHANGES_REQUESTED** [review]

File: lib/vue/src/components/resourceList/CResourceList.vue (hunk 11 above)
Comment (EmmyMay) [reply lib/vue/src/components/resourceList/CResourceList.vue]: putting it in span means I have to add a block class to it which is the same as having a div imo

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 15]
```diff
@@ -342,10 +384,11 @@ export default {
 
     hasNextPage () {
       return !!this.pagination.nextPage
+
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Lint please, no need for this extra line and spaces

File: lib/vue/src/components/resourceList/CResourceList.vue [hunk 16]
```diff
@@ -342,10 +384,11 @@ export default {
 
     hasNextPage () {
       return !!this.pagination.nextPage
+ 
     },
 
     showFooter () {
-      return !(this.hideTotal && this.hidePagination)
+      return !(this.hideTotal && this.hidePagination && !this.hidePerPageOption)
```
Comment (Fajfa) [inline lib/vue/src/components/resourceList/CResourceList.vue]: Fix this please

**Review by Fajfa — CHANGES_REQUESTED** [review]

**Comment by katrinDY** [comment]
One small thing in reporter, else all good:

Screenshot 2023-11-29 at 13 10 59

**Pull request merged**